style/git_and_github.qmd

```diff
@@ -1,17 +1,88 @@
+-   All code should be committed via pull request (PR) to `main`
+
+::: {.callout-important appearance="default" icon="true"}
+**Never** push to `main`!
+:::
 
 Workflow for writing code with Git and GitHub:
 
-* If you haven't already file an issue that describes what you are doing- whether it be fixing a bug, adding a feature, or something else
-* Make a new branch to work on that feature. GitHub offers a button to do this automatically on the issue page, on the right hand side (see image)
-* Work on the branch, pushing your code regularly to GitHub so it can be run and inspected when you are not around
-* When it's ready merge the branch to main. If you used the create branch button on GitHUb this will automatically close the issue. If the issue is not automatically closed, then close it manually
+## Create an issue
+
+If you haven't already, file an issue that describes what you are doing - whether it be fixing a bug, adding a feature, or something else.
+Issues help to keep track of both the work to be done, and the work that has been done.
+
+You should ensure that the issue is as detailed as possible as it may need to be picked up by someone else.
+Even if only you intend to work on the issue, over time it can be easy to forget things if there isn't enough information.
+They can also help you to remember what changes you have made to your code, and why you had to make these changes, which can be useful if you need to provide change logs when you release your code.
+
+::: aside
+Issues are a good record to explain what a PR is for.
+It's also a useful area for input from other members of the team.
+:::
+
+## Create a branch
+
+Once the issue is created and you have decided that you are going to work on it, first assign yourself to the issue in GitHub.
+This is an indication to others and helps to prevent multiple people from independently working on the same issue.
+
+Once you have assigned yourself, you must make a new branch to work on that feature.
+GitHub offers a button to do this automatically on the issue page, on the right-hand side.
+
+[![Screenshot links to specific GitHub help page](branch_img.png){fig-alt="Screenshot of GitHub with the hyperlink to Create a branch highlighted under the category of Development."}](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-a-branch-for-an-issue)
+
+Locally you can then work on the branch, pushing your code regularly to GitHub so it can be run and inspected when you are not around.
+
+When you think that your changes are ready to be merged, it's time to create a PR and request a code review.
+
+::: aside
+If you used the `create branch` button on GitHub it will automatically close the issue when merged.
+:::
+
+## Reviewing a PR
+
+When you create a PR, you must do two things:
+
+1.  immediately make someone an assignee - this is the person who will merge the PR.
+    Typically, this should be the person creating the PR (you)
```

Comment:
Is it more typical that the PR creator will merge the PR or the branch owner will merge the PR?